sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java

```diff
@@ -0,0 +1,186 @@
+package org.dbsp.sqlCompiler.compiler.sql.simple;
+
+import org.dbsp.sqlCompiler.compiler.sql.SqlIoTest;
+import org.junit.Test;
+
+public class TrigonometryTests extends SqlIoTest {
+    // Tested using Postgres 15.2
+    @Test
+    public void testSin() {
+        this.qs(
+        "SELECT sin(null);\n" +
+                " sin \n" +
+                "-----\n" +
+                "NULL\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(0);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(0.25);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0.24740395925452294\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(0.5);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0.479425538604203\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(0.75);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0.6816387600233341\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(1);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0.8414709848078965\n" +
+                "(1 row)\n" +
+                "\n" +
+                "SELECT sin(3.14);\n" +
+                " sin \n" +
+                "-----\n" +
+                " 0.0015926529164868282\n" +
+                "(1 row)"
+        );
+    }
+
+    // Tested using Postgres 15.2
+    @Test
+    public void testSinDouble() {
+        this.qs(
+                "SELECT sin(CAST(0 AS DOUBLE));\n" +
```

Comment:
this works, but you can get a double by writing `e0` at the end. `3.14e0` is a double, while `3.14` is a decimal.